builtin/logical/pki/path_intermediate_test.go

```diff
@@ -0,0 +1,74 @@
+package pki
+
+import (
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/vault/helper/certutil"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func TestSetSignedIntermediate(t *testing.T) {
+	storage := &logical.InmemStorage{}
+	config := logical.TestBackendConfig()
+	config.StorageView = storage
+
+	b := Backend()
+	_, err := b.Setup(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Put cert bundle in inmem storage
+	privateCertPEM, err := ioutil.ReadFile("test-fixtures/cakey.pem")
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	cb := &certutil.CertBundle{}
+	cb.PrivateKey = string(privateCertPEM)
+	cb.PrivateKeyType = certutil.RSAPrivateKey
+
+	bundleEntry, err := logical.StorageEntryJSON("config/ca_bundle", cb)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = storage.Put(bundleEntry)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	certValue, err := ioutil.ReadFile("test-fixtures/cacert.pem")
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	req := &logical.Request{
+		Operation: logical.UpdateOperation,
+		Storage:   storage,
+	}
+
+	fd := &framework.FieldData{
+		Raw: map[string]interface{}{
+			"certificate": certValue,
+		},
+		Schema: pathSetSignedIntermediate(b).Fields,
+	}
+
+	resp, err := b.pathSetSignedIntermediate(req, fd)
```

Comment:
On a second thought, I think I should be using `b.HandleRequest(req)` here instead, with the appropriate req.Path and req.Data. Is that correct @jefferai?